### PR TITLE
User/rahulaga/fix ms build verion for dotnet31

### PR DESCRIPTION
**Pull request by rahulaga-msft**
updated to use MSBuild 16.7 for dotnet core 3.1 applications

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3359) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3359) before we can accept your contribution.<br/><br/>:x: rahulaga-msft [sign now](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3359)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/projectkudu/kudu?pullRequest=3359) it.</sub>

**Comment by patelchandni** [comment]
Looks good to me. @khkh-ms and @suwatch please review it as well.

**Comment by suwatch** [comment]
please squash into one commit and rebase with master - thanks

File: Kudu.Core/Infrastructure/PathUtils/PathWindowsUtility.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ internal override string ResolveMSBuild15Dir()
             return probPaths.FirstOrDefault(path => Directory.Exists(path));
         }
 
-        internal override string ResolveMSBuild16Dir()
+        internal override string ResolveMSBuild16Dir(bool IsDotNet31Project=false)
```
Comment (khkh-ms) [inline Kudu.Core/Infrastructure/PathUtils/PathWindowsUtility.cs:124]: Trying to understand why parameter default value was used? `ResolveMSBuild16Dir` is only referenced twice. Can we just pass the right value from both method calls?

**Review by khkh-ms — APPROVED** [review]
Looks good to me.

File: Kudu.Core/Infrastructure/PathUtils/PathWindowsUtility.cs (hunk 1 above)
Comment (rahulaga-msft) [reply Kudu.Core/Infrastructure/PathUtils/PathWindowsUtility.cs:124]: This I did to ensure `no regression issue` and making sure that it points to `MSBuild 16.7` only for `dotnet 3.1` project. As I see from SiteBuilderFactory, this path can be taken for another target framework as well (anything not `net5.0` and `net6.0`) hence this default to `false` will ensure it explicitly takes that path only for `3.1` project

**Pull request merged**